src/dd_header.F90

```diff
@@ -0,0 +1,109 @@
+module dd_header
+
+  use dict_header,      only: DictIntInt
+  use mesh_header,      only: RegularMesh
+  use particle_header,  only: Particle, ParticleBuffer
+
+  implicit none
+  private
+  public :: deallocate_dd
+
+  type, public :: dd_type
+
+    ! Domain mesh information
+    type(RegularMesh), pointer :: mesh => null()
```

Comment:
Excuse me, @paulromano , after changing this to `type(RegularMesh) :: mesh`, how to deallocate this `mesh` in `subroutine deallocate_dd(this)`? Thanks.